<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>860</width>
    <height>631</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>QR Text Encryptor</string>
  </property>
  <property name="windowIcon">
   <iconset resource="../qrc.qrc">
    <normaloff>:/img/qrencryptor.png</normaloff>:/img/qrencryptor.png</iconset>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout_6">
    <item row="0" column="0">
     <widget class="QFrame" name="frame_3">
      <property name="minimumSize">
       <size>
        <width>370</width>
        <height>0</height>
       </size>
      </property>
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
      <layout class="QGridLayout" name="gridLayout_5">
       <item row="3" column="0">
        <widget class="QLabel" name="label_7">
         <property name="text">
          <string>Iterations:</string>
         </property>
        </widget>
       </item>
       <item row="0" column="0">
        <widget class="QLabel" name="label_4">
         <property name="text">
          <string>Encryption:</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1" colspan="2">
        <widget class="QComboBox" name="comboBox_errorCorrectionLevel">
         <item>
          <property name="text">
           <string>LOW</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>MEDIUM</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>QUARTILE</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>HIGH</string>
          </property>
         </item>
        </widget>
       </item>
       <item row="0" column="1" colspan="2">
        <widget class="QComboBox" name="comboBox">
         <item>
          <property name="text">
           <string>AES-256-GCM</string>
          </property>
         </item>
        </widget>
       </item>
       <item row="3" column="2">
        <widget class="QLabel" name="label_iterations">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="maximumSize">
          <size>
           <width>70</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="text">
          <string>0</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="label_5">
         <property name="text">
          <string>Error Correction Level:</string>
         </property>
        </widget>
       </item>
       <item row="3" column="1">
        <widget class="QSpinBox" name="spinBox_iterations">
         <property name="minimum">
          <number>100000</number>
         </property>
         <property name="maximum">
          <number>10000000</number>
         </property>
         <property name="singleStep">
          <number>100000</number>
         </property>
         <property name="value">
          <number>3000000</number>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="0" column="1">
     <widget class="QFrame" name="frame_2">
      <property name="minimumSize">
       <size>
        <width>370</width>
        <height>0</height>
       </size>
      </property>
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
      <layout class="QGridLayout" name="gridLayout">
       <item row="2" column="2">
        <spacer name="horizontalSpacer_4">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="label_6">
         <property name="text">
          <string>Strength:</string>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QLabel" name="label_strength">
         <property name="minimumSize">
          <size>
           <width>16</width>
           <height>16</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>16</width>
           <height>16</height>
          </size>
         </property>
         <property name="styleSheet">
          <string notr="true">background-color:red;</string>
         </property>
         <property name="text">
          <string/>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="label_Pass_2">
         <property name="text">
          <string>Repeat:</string>
         </property>
        </widget>
       </item>
       <item row="0" column="0">
        <widget class="QLabel" name="label_Pass_1">
         <property name="text">
          <string>Passphrase:</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1" colspan="4">
        <widget class="QLineEdit" name="lineEdit_Pass_1">
         <property name="echoMode">
          <enum>QLineEdit::Password</enum>
         </property>
        </widget>
       </item>
       <item row="1" column="1" colspan="4">
        <widget class="QLineEdit" name="lineEdit_Pass_2">
         <property name="echoMode">
          <enum>QLineEdit::Password</enum>
         </property>
        </widget>
       </item>
       <item row="2" column="3">
        <widget class="QCheckBox" name="checkBox_visiblePass">
         <property name="text">
          <string>View Password</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="1" column="0" colspan="2">
     <widget class="QTabWidget" name="tabWidget">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="tab_encoding">
       <attribute name="title">
        <string>Text Encryption</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_2">
        <item row="0" column="1">
         <widget class="QLabel" name="label">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>Description:</string>
          </property>
         </widget>
        </item>
        <item row="1" column="4">
         <widget class="QToolButton" name="toolButton_clear_encoding">
          <property name="text">
           <string>...</string>
          </property>
          <property name="icon">
           <iconset theme="edit-clear">
            <normaloff>.</normaloff>.</iconset>
          </property>
         </widget>
        </item>
        <item row="3" column="1">
         <widget class="QLabel" name="label_2">
          <property name="text">
           <string>Encrypted:</string>
          </property>
         </widget>
        </item>
        <item row="0" column="2">
         <widget class="QLineEdit" name="lineEdit_description">
          <property name="text">
           <string>Encrypted QR</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QLabel" name="label_3">
          <property name="text">
           <string>Plain text:</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1" colspan="4">
         <widget class="QPlainTextEdit" name="plainTextEdit_PlainTextToEncode">
          <property name="styleSheet">
           <string notr="true">font: 10pt &quot;Courier 10 Pitch [bitstream]&quot;;</string>
          </property>
         </widget>
        </item>
        <item row="4" column="1" colspan="4">
         <widget class="QPlainTextEdit" name="plainTextEdit_Encoded">
          <property name="maximumSize">
           <size>
            <width>16777215</width>
            <height>100</height>
           </size>
          </property>
          <property name="styleSheet">
           <string notr="true">background-color: rgb(245, 255, 245);</string>
          </property>
          <property name="readOnly">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item row="0" column="3" colspan="2">
         <widget class="QComboBox" name="comboBox_textSize">
          <item>
           <property name="text">
            <string>8</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>10</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>12</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>14</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>16</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>20</string>
           </property>
          </item>
         </widget>
        </item>
        <item row="1" column="2" colspan="2">
         <spacer name="horizontalSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>339</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="5" column="3" colspan="2">
         <widget class="QPushButton" name="pushButton_encrypt">
          <property name="text">
           <string>Encrypt...</string>
          </property>
          <property name="icon">
           <iconset resource="../qrc.qrc">
            <normaloff>:/img/logo.png</normaloff>:/img/logo.png</iconset>
          </property>
         </widget>
        </item>
        <item row="5" column="1" colspan="2">
         <spacer name="horizontalSpacer_3">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="0" column="0" rowspan="6">
         <widget class="QFrame" name="frame">
          <property name="minimumSize">
           <size>
            <width>350</width>
            <height>370</height>
           </size>
          </property>
          <property name="styleSheet">
           <string notr="true">background-color: gray;</string>
          </property>
          <property name="frameShape">
           <enum>QFrame::StyledPanel</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Raised</enum>
          </property>
          <layout class="QGridLayout" name="gridLayout_4">
           <item row="1" column="0">
            <spacer name="verticalSpacer">
             <property name="orientation">
              <enum>Qt::Vertical</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>20</width>
               <height>40</height>
              </size>
             </property>
            </spacer>
           </item>
           <item row="0" column="0">
            <widget class="QLabel" name="label_qr">
             <property name="minimumSize">
              <size>
               <width>300</width>
               <height>350</height>
              </size>
             </property>
             <property name="styleSheet">
              <string notr="true">background-color: white;</string>
             </property>
             <property name="text">
              <string/>
             </property>
             <property name="scaledContents">
              <bool>false</bool>
             </property>
             <property name="alignment">
              <set>Qt::AlignCenter</set>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_decoding">
       <attribute name="title">
        <string>Text Decryption</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_3">
        <item row="1" column="2">
         <widget class="QPushButton" name="pushButton_Decode">
          <property name="text">
           <string>Decode from Base64</string>
          </property>
          <property name="icon">
           <iconset resource="../qrc.qrc">
            <normaloff>:/img/b64.png</normaloff>:/img/b64.png</iconset>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <spacer name="horizontalSpacer_2">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>267</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="1" column="0">
         <widget class="QToolButton" name="toolButton_clear_decoding">
          <property name="text">
           <string>...</string>
          </property>
          <property name="icon">
           <iconset theme="edit-clear">
            <normaloff>.</normaloff>.</iconset>
          </property>
         </widget>
        </item>
        <item row="0" column="0" colspan="3">
         <widget class="QPlainTextEdit" name="plainTextEdit_TextToDecode">
          <property name="styleSheet">
           <string notr="true">font: 10pt &quot;Courier 10 Pitch [bitstream]&quot;;</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>860</width>
     <height>23</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionPrint"/>
    <addaction name="separator"/>
    <addaction name="actionE_xit"/>
   </widget>
   <widget class="QMenu" name="menuEdit">
    <property name="title">
     <string>Edit</string>
    </property>
    <addaction name="actionCopy_QR_to_Clipboard"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionAbout"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuEdit"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionE_xit">
   <property name="icon">
    <iconset theme="application-exit">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>E&amp;xit</string>
   </property>
  </action>
  <action name="actionPrint">
   <property name="icon">
    <iconset theme="printer">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>Print...</string>
   </property>
  </action>
  <action name="actionCopy_QR_to_Clipboard">
   <property name="text">
    <string>Copy QR to Clipboard</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>About</string>
   </property>
  </action>
 </widget>
 <tabstops>
  <tabstop>lineEdit_Pass_1</tabstop>
  <tabstop>plainTextEdit_PlainTextToEncode</tabstop>
  <tabstop>plainTextEdit_TextToDecode</tabstop>
  <tabstop>pushButton_Decode</tabstop>
 </tabstops>
 <resources>
  <include location="../qrc.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>actionE_xit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>429</x>
     <y>315</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
